cache/decoratorCache.go

```diff
@@ -0,0 +1,162 @@
+
+func getUserCache(identity, service string, logger jsonlog.Logger) interface{} {
+	var cacheData interface{} = nil
+	rtn, err := mainClient.Get(getUserKey(identity, service)).Bytes()
+	if err != nil {
+		_ = logger.Error(fmt.Sprintf("No cache found on Redis for the '%v' despite it has marked as valid\n", identity), err)
+		return nil
+	} else {
+		err := json.Unmarshal(rtn, &cacheData)
+		if err != nil {
+			_ = logger.Error(fmt.Sprintf("Unable to unmarshal cache data for the key '%v'\n", identity), err)
+			return nil
+		}
+		return cacheData
+	}
+}
+
+func createUserCache(identity, service string, data interface{}, logger jsonlog.Logger) error {
+	if userHasCacheForService(identity, service, logger) {
+		_ = logger.Warning(fmt.Sprintf("User %v has already a cache attributed for service '%v'", identity, service), nil)
+		return nil
+	}
+	content, newErr := json.Marshal(data)
+	if newErr != nil {
+		_ = logger.Error("Unable to marshal API content to create cache.", nil)
+		return newErr
+	}
+	userKey := getUserKey(identity, service)
+	err := mainClient.Append(userKey, string(content))
+	if err.Err() == nil {
+		mainClient.Expire(userKey, cacheExpireTime)
+	} else {
+		_ = logger.Error(fmt.Sprintf("Unable to append content for the key '%v'.", userKey), err.Err())
+	}
+	return err.Err()
+}
+
+func deleteUserCache(identity, service string) {
+	mainClient.Del(getUserKey(identity, service))
+}
+
+// getFunc allows us to intercept the current data flow from the route and
+// manipulate it to retrieve data or directly return data from the cache if
+// there is one.
+func (_ UsersCache) getFunc(hf routes.HandlerFunc) routes.HandlerFunc {
+	return func(writer http.ResponseWriter, request *http.Request, args routes.Arguments) (int, interface{}) {
+		logger := jsonlog.LoggerFromContextOrDefault(request.Context())
+		_, userDataPresent := args[users.AuthenticatedUser]
+		_, dbAvailable := args[db.Transaction]
+		if !userDataPresent || !dbAvailable {
+			_ = logger.Error("Unable to retrieve user's or database's information while trying to get cache.", nil)
+			return hf(writer, request, args)
+		}
+		service := request.URL.String()
+		identity, identityErr := getAwsAccIdFromRaw(args)
+		if identityErr != nil {
+			_ = logger.Error("Unable to retrieve user's AWS account id", identityErr.Error())
+			return hf(writer, request, args)
+		}
+		if userHasCacheForService(identity, service, logger) {
+			retrieveCache := getUserCache(identity, service, logger)
+			if retrieveCache == nil {
+				_ = logger.Warning(fmt.Sprintf("Unable to retrieve cache for key %v, skipping it to avoid panic or error. The cache has been, also, deleted.\n", getUserKey(identity, service)), nil)
+				deleteUserCache(identity, service)
+			} else {
+				return http.StatusOK, retrieveCache
+			}
+		}
+		status, routeData := hf(writer, request, args)
+		if status == http.StatusOK {
+			_ = createUserCache(identity, service, routeData, jsonlog.LoggerFromContextOrDefault(request.Context()))
```

Comment:
Why not passing `logger` instead of `jsonlog.LoggerFromContextOrDefault(request.Context())` ?